classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -94,10 +94,6 @@ public function up(): void
             });
         }
 
-        // Use nulls instead of 0 for assoc_type/id in user_settings
-        DB::table('user_settings')->where('assoc_type', 0)->update(['assoc_type' => null]);
-        DB::table('user_settings')->where('assoc_id', 0)->update(['assoc_id' => null]);
```

Comment:
I've removed this as it's probably the origin of the duplicated records.

Reply:
@jonasraoni, I think we should standardize on `null` instead of `0` *everywhere* we're currently using `0` as a placeholder ID. As we start declaring `FOREIGN KEY` relationships formally, we won't be able to get away with this looseness any more. (That change will come to `assoc_type` / `assoc_id` relationships last, because DBMSs won't understand them at the DDL level, but I think it's important to be consistent anyway as we roll out code that's pickier about `0` vs. `null` in PHP.)

FWIW, this change was added [here](https://github.com/pkp/pkp-lib/pull/6347/files) by @Vitaliy-1.

Reply:
That's right, but the problem here is that the table was updated, together with all its records, but the code stills using zeroes, then whenever a user updates its data, things get duplicated (older record with null + new record with 0).

But in fact, I just removed this to speed up the migration, given that these fields will be removed by my PR later on.

Reply:
Sooo, I think these two lines should be also removed from the `stable-3_3_0` branch. If you agree, I can create an issue + PR for it.

Reply:
That makes sense -- please go ahead!